sql/src/main/java/io/crate/execution/dsl/projection/UpdateProjection.java

```diff
@@ -118,6 +136,14 @@ public ProjectionType projectionType() {
         return visitor.visitUpdateProjection(this, context);
     }
 
+    @Override
+    public List<? extends Symbol> outputs() {
+        if (output == null) {
+            return super.outputs();
```

Comment:
I don't like it, to return a value of a super class when the current value is null. One solution would be to not extend from DMLProjection and just handle the return types accordingly here.

Reply:
+1 for not extending DMLProjection